src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -413,6 +444,73 @@ def read_file(
                 result_df = pd.concat([result_df, df], ignore_index=True)
             result_df.sf_types = result_df_sf_types
             return result_df
+        elif file_format == "json":
+            infer_schema_opt = options.get("INFER_SCHEMA", False)
+
+            result_df = TableEmulator()
+            result_df_sf_types = {}
+
+            if not infer_schema_opt:
+                # snowflake output the value as raw string
+                for i in range(len(schema)):
+                    column_name = analyzer.analyze(schema[i])
+                    column_series = ColumnEmulator(
+                        data=None, dtype=object, name=column_name
+                    )
+                    column_series.sf_type = ColumnType(
+                        schema[i].datatype, schema[i].nullable
+                    )
+                    result_df[column_name] = column_series
+                    result_df_sf_types[column_name] = column_series.sf_type
+
+                for local_file in local_files:
+                    with open(local_file) as file:
+                        content = json.load(file)
+                        df = pd.DataFrame({result_df.columns[0]: [content]})
```

Comment:
Does this mean we ignore all the other columns? Or can there only be one column in the schema if `infer_schema_opt` is False?

Reply:
it's the latter one that there should only be 1 column if not inferring schema
added an assertion at the beginning.